felix/dataplane/linux/bpf_route_mgr.go

```diff
@@ -610,12 +629,15 @@ func (m *bpfRouteManager) onRouteUpdate(update *proto.RouteUpdate) {
 
 func (m *bpfRouteManager) onRouteRemove(update *proto.RouteRemove) {
 	cidr := ip.MustParseCIDROrIP(update.Dst)
-	if !m.ipv6Enabled {
+	if m.ipFamily == proto.IPVersion_IPV4 {
 		if _, ok := cidr.(ip.V4CIDR); !ok {
 			return
 		}
+	} else {
+		if _, ok := cidr.(ip.V6CIDR); !ok {
+			return
```

Comment:
It feels like it almost asks for making `bpfRouteManager` a generic parametrized with `ip.V4CIDR` or `ip.CIDR has `Version() uint8` method so we would not even need to know the type (my bad :sweat_smile: )